src/name.rs

```diff
@@ -77,20 +77,28 @@ impl<'a> From<DNSNameRef<'a>> for DNSName {
 #[derive(Clone, Copy)]
 pub struct DNSNameRef<'a>(untrusted::Input<'a>);
 
+/// A marker struct indicating that a `DNSNameRef` could not built from the provided input.
+///
+/// This struct is voluntarily kept separate from the `Error` enum to emulate the behavior of the
+/// still unstable `TryFrom` trait, leaving the calling code free to take corrective action or
+/// convert `NotADNSName` into an error type it knows about.
+#[derive(Clone, Copy)]
+pub struct NotADNSName<'a>(untrusted::Input<'a>);
```

Comment:
Please add this:

```rust
impl fmt::Display for NotADNSName {
	fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
		write!(f, "{:?}", self)
	}
}

#[cfg(feature = "std")]
impl ::std::error::Error for NotADNSName {}

Reply:
I assume this is to `no_std`-proof the code, shouldn't also `impl Display`s be gated in this case?